app/src/main/kotlin/at/bitfire/davdroid/repository/SyncOnCollectionsChangeListener.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright © All Contributors. See LICENSE and AUTHORS in the root directory for details.
+ */
+
+package at.bitfire.davdroid.repository
+
+import android.accounts.Account
+import at.bitfire.davdroid.sync.worker.SyncWorkerManager
+import dagger.Binds
+import dagger.Module
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import dagger.multibindings.IntoSet
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import java.util.logging.Logger
+import javax.inject.Inject
+
+/**
+ * Enqueues a sync worker after a short delay when the collection list changes.
+ */
+class SyncOnCollectionsChangeListener @Inject constructor(
+    private val workerManager: SyncWorkerManager,
+    private val logger: Logger
+): DavCollectionRepository.OnChangeListener {
+
+    var delayedOneTimeSyncWorkerJob: Job? = null
+
+    override fun onCollectionsChanged(account: Account?) {
+        account?.let {
+            // Start sync after a short delay to avoid multiple syncs in a short time when multiple
+            // collections change quickly, e.g. at collection refresh or users first time setup.
+            delayedOneTimeSyncWorkerJob?.cancel()
+            delayedOneTimeSyncWorkerJob = CoroutineScope(Dispatchers.IO).launch {
+                delay(7000)
```

Comment:
Just a small comment, shouldn't this be set to some constant or something just in case we want to modify it at some point in the future?

Reply:
I don't see an advantage in doing that at this moment. With the comment it's quite clear what's happening, so no need to use a descriptive variable/constant name and we can easily change the value in place.

If, some time in the future, we start dealing with multiple delays (for whatever reason I can't think of at the moment) in this same class, then I think it would make sense to have a central place to configure all the delays (likely the companion object).

See also principles of YAGNI and KISS.